src/tools/compiletest/src/main.rs

```diff
@@ -591,6 +595,24 @@ fn collect_tests_from_dir(
     let build_dir = output_relative_path(config, relative_dir_path);
     fs::create_dir_all(&build_dir).unwrap();
 
+    if config.only_modified {
+        let Ok(Some(files)) = get_git_modified_files(Some(dir), &vec!["rs", "stderr", "fixed"]) else { return Ok(()); };
+        // Add new test cases to the list, it will be convenient in daily development.
+        let Ok(Some(untracked_files)) = get_git_untracked_files(None) else { return Ok(()); };
+        let mut all_paths = files;
+        all_paths.extend(untracked_files);
+        let mut full_paths: Vec<PathBuf> =
+            all_paths.iter().map(|f| fs::canonicalize(&f).unwrap().with_extension("rs")).collect();
+        full_paths.dedup();
```

Comment:
```suggestion
        let all_paths = {
            let mut all_paths = files;
            all_paths.extend(untracked_files);
            all_paths
        };
        let full_paths = {
            let mut full_paths: Vec<PathBuf> =
                all_paths.into_iter().map(|f| fs::canonicalize(&f).unwrap().with_extension("rs")).collect();
            full_paths.dedup();
            full_paths
        };
```

This is a very OCD nit, but the less mutability there is, the better is for me (absolutely not a blocker)

Reply:
Use this to concate two `vec` seems better?

```rust
let all_paths = [&files[..], &untracked_files[..]].concat();
```

Reply:
If it works, then yes.